src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs

```diff
@@ -469,7 +645,7 @@ private async Task<bool> SynchronizeBatchAsync()
                     break;
             }
 
-            List<Block> blocks = await this.blockRepository.GetBlocksAsync(blockHashes);
+            List<Block> blocks = this.blockRepository.GetBlocksAsync(blockHashes).GetAwaiter().GetResult();
             int availableBlocks = blocks.FindIndex(b => (b == null));
             if (availableBlocks < 0)
```

Comment:
This is checking for a negative `availableBlock` value - is this right ?

Reply:
Yes. If `FindIndex` finds none of the blocks that are null, it return -1, which indicated that the blocks are all usable.

Reply:
Ok, thanks.